.github/workflows/pre-release.yml

```diff
@@ -18,7 +14,8 @@ jobs:
       matrix:
         edition: [oss, ee]
     outputs:
-      version: ${{ steps.version-properties.outputs.version }}
+      ee_version: ${{ steps.version-properties.outputs.ee_version }}
+      oss_version: ${{ steps.version-properties.outputs.oss_version }}
```

Comment:
A matrix in this job does not guarantee uninterrupted thread/context across the following jobs.
The only way to be sure that jobs downstream use the correct version is to create two separate outputs. This requires a bit more verbose conditional logic in the downstream jobs, unfortunately.